gen3users/user_security.py

```diff
@@ -107,58 +109,80 @@ def _check_yaml(
 
         logging.info(f"========= Scanning file for {phsid} in open projects =========")
         found = False
-        # Track down open_data_reader
+
+        # Track down open data policies (anonymous or all_users)
         policies = content[auth_meth]["policies"]
+        open_policies = []
+
         for i in range(len(policies)):
-            if (
-                policies[i]["id"] in content[auth_meth]["anonymous_policies"]
-                or policies[i]["id"] in content[auth_meth]["all_users_policies"]
-            ):
-                policy = policies[i]
-                break
-
-        # Check if phsid is hardcoded into the open data reader resource path
-        if (
-            phsid in policy["resource_paths"]
-            or "/programs/" + phsid in policy["resource_paths"]
-        ):
-            logging.info("PHSID Directly in Open Data Reader Resource Path")
-
-        # Check if phsid is in one of the resource paths
-        else:
-            resources = content[auth_meth]["resources"]
-            open_resources = policy["resource_paths"]
-            for i in range(len(open_resources)):
-                resource = resources
-                path = open_resources[i].split("/")
-                for j in range(len(path)):
-                    for k in range(len(resource)):
-                        if resource[k]["name"] == path[j]:
-                            try:
-                                resource = resource[k]["subresources"]
-                                break
-                            except KeyError:
-                                break
-
-                locs = []
-                for j in range(len(resource)):
-                    if resource[j]["name"] == "projects":
-                        resource = resource[j]["subresources"]
-                        for k in range(len(resource)):
-                            if resource[k]["name"] == phsid:
-                                found = True
-                                anything_found = True
-                                location = "resources" + "/".join(path)
-                                locs.append(location)
-
-            if found == True:
-                logging.info(f"{phsid} found in {*locs,} listed as open\n")
+            # Some user.yaml files don't have these specified so handle with try block
+            try:
+                bool1 = policies[i]["id"] in content[auth_meth]["anonymous_policies"]
+            except KeyError:
+                bool1 = False
+
+            try:
+                bool2 = policies[i]["id"] in content[auth_meth]["all_users_policies"]
+            except KeyError:
+                bool2 = False
+            # If a policy is in either all_users_policies or anonymous_policies it is open
+            if bool1 or bool2:
+                open_policies.append(policies[i])
+
+        # Iterate through open policies
+        for policy in open_policies:
```

Comment:
This can probably also be a separate method. In general, a lot of the code inside this method could be extracted out into separate methods to clean things up a little and separate concerns.